src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs

```diff
@@ -153,6 +153,13 @@ private bool TryGetTextForSymbol(SyntaxToken token, SemanticModel semanticModel,
                 symbol = ((ILocalSymbol)symbol).Type;
             }
 
+            // Range variable: use the type
+            if (symbol is IRangeVariableSymbol)
+            {
+                var info = semanticModel.GetSpeculativeTypeInfo(token.SpanStart, SyntaxFactory.ParseName(symbol.Name), SpeculativeBindingOption.BindAsExpression);
+                symbol = info.Type;
```

Comment:
This PR was originally about searching for the names of properties, but @dpoeschl pointed out that I might as well make range variables work as well. It's not interesting to search MSDN for names of range variables, so we search for the type.

The API requires that we speculatively bind range variables to get their types-see CSharpMethodExtractor.Analyzer.GetRangeVariableType.